src/examples/poi-crosschecker/proto/NPOIMessage.proto

```diff
@@ -9,4 +9,5 @@ message NPOIMessage {
     string sender = 5;
     string subgraph = 6;
     string nPOI = 7;
+    string signature = 8;
```

Comment:
I think we should move signature outside of the message content/payload. Other fields like nonce, timestamp,... too. Could we declare nested message types? something like 
```
message GossipMessage {
  int64 nonce = 1;
  string signature = 2;
   ...
   oneof content {
        GossipNPOI nPOI = 3;
   }
}

message GossipNPOI {
    int64 nonce         = 1;
    bytes metadataNPOI       = 2;
    string msg_type = 3;
}
```
what do you think?

Reply:
I completely agree. Like the previous point, this is something I feel would be right to do as part of the refactoring, and I think ultimately we would want 3 levels of a message - 
```
1. **Outer metadata** - the metadata that Waku wraps the messages in (from which we'll only use the `timestamp` for now).
    2. **Inter metadata** - the metadata that each Graphcast message will have and that will be defined and handled by the SDK, regardless of the specific Radio logic, stuff like `nonce` and `signature` as you mentioned.
        3. **The content/payload** specific to the Radio, this one is user-defined.
```
Lmk if that sounds good 🙌🏻

Reply:
sounds great! Agree with the three levels you have described:)